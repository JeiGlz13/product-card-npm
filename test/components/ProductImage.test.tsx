import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/product';

describe('Pruebas en ProductImage', () => { 
    test('Debe mostar el ProductImage correctamente', () => { 
        const wrapper = renderer.create(
            <ProductImage img='https://MyImage.jpg' />
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
     });

     test('debe mostar el componente con el image del context', () => { 
        const wrapper = renderer.create(
            <ProductCard product={product2} >
                {
                    ()=>(
                        <ProductImage/>
                    )
                }
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
      });
 })