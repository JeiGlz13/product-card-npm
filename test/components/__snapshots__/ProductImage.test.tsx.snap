// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ProductImage Debe mostar el ProductImage correctamente 1`] = `
<div
  className="lg:relative w-7/12 px-4 py-4 flex"
>
  <img
    alt="Product"
    className="w-full rounded-3xl undefined"
    src="https://MyImage.jpg"
  />
</div>
`;

exports[`Pruebas en ProductImage debe mostar el componente con el image del context 1`] = `
<div
  className="mt-8 w-96 mx-2 md:p-4 p-2 bg-white rounded-3xl undefined"
>
  <div
    className="lg:relative w-7/12 px-4 py-4 flex"
  >
    <img
      alt="Product"
      className="w-full rounded-3xl undefined"
      src="./coffee-mug.png"
    />
  </div>
</div>
`;
